### PR TITLE
refactor (js/contentscript) manage state by methods

### PR DESCRIPTION
**Pull request by taka-sho**

### CODE REVIEW

File: app/js/contentscript.js [hunk 1]
```diff
@@ -1,5 +1,6 @@
 /* @flow */
 declare var chrome: any
+let state = {}
```
Comment (9renpoto) [inline app/js/contentscript.js]: why state is global vars

Comment (heavenshell) [inline app/js/contentscript.js]: Why outside of class?
Prefer defined at class property?

File: app/js/contentscript.js [hunk 2]
```diff
@@ -10,36 +11,58 @@ class Provider {
...
+          })
+          this.renderBadge('OFF')
           return
         }
         this.readyState()
       })
     })
   }
+  setState (data: object): void {
+    if (data) {
```
Comment (heavenshell) [inline app/js/contentscript.js]: `data` is nullable?

File: app/js/contentscript.js [hunk 3]
```diff
@@ -10,36 +11,58 @@ class Provider {
...
+      state = data
+      return
+    }
+    state = {}
```
Comment (heavenshell) [inline app/js/contentscript.js]: IMO, you don't need clear state here.
If you want to clear state, you may provide `clear()` method.

File: app/js/contentscript.js [hunk 4]
```diff
@@ -10,36 +11,58 @@ class Provider {
...
+  }
+  getState (): object {
+    if (state) {
+      return state
+    }
+    return {}
```
Comment (heavenshell) [inline app/js/contentscript.js]: I think this line never reached.

Comment (shoota) [inline app/js/contentscript.js]: why return empty Object? where is truth state...

File: app/js/popup.js [hunk 5]
```diff
@@ -5,17 +5,25 @@ declare var chrome: any
...
   render (): void {
     chrome.tabs.query({ active: true, currentWindow: true }, tabs => {
-      chrome.tabs.sendMessage(tabs[0].id, { content: 'fetchCookie' }, response => {
-        if (!response || !response.data) {
-          throw new Error("couldn't recieve page data")
+      chrome.tabs.sendMessage(
+        tabs[0].id,
+        { content: 'fetchCookie' },
+        response => {
+          if (!response || !response.data) {
```
Comment (shoota) [inline app/js/popup.js]: prefer to write expression as `if (response && response.data)` for me

**Pull request merged**